### PR TITLE
Try speeding up yarn install in CI using yarn install --frozen-lockfile

### PR DESCRIPTION
**Pull request by jywarren**
https://stackoverflow.com/questions/58482655/what-is-the-closest-to-npm-ci-in-yarn

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
         bundler-cache: true # runs 'bundle install' and caches installed gems automatically (not working?)
     - uses: ./.github/actions/setup-test-environment
     - name: Install JavaScript dependencies with Yarn
-      run: yarn check || yarn install;
+      run: yarn check || yarn install --frozen-lockfile;
```
Comment (Sagarpreet) [inline .github/workflows/tests.yml:43]: I think we do not add modules so frequently, right? Can we cache node_modules folder somehow and do `yarn install` only if package.json or yarn.lock changes to speed things up?

Comment (jywarren) [reply .github/workflows/tests.yml:43]: I looked this up and found this was the most common and "standard" way to speed up yarn installs, but yeah, i think we can use one of these:

https://dev.to/mpocock1/how-to-cache-nodemodules-in-github-actions-with-yarn-24eh

https://github.com/actions/cache/blob/main/examples.md#node---yarn

Let's try in a different PR? Thanks!

**Comment by jywarren** [comment]
Lets compare to times here: 

https://github.com/publiclab/plots2/actions/runs/534731427

image

**Comment by jywarren** [comment]
OK, here's one run: 

image

I'll re-run again to check consistency.

**Comment by jywarren** [comment]
wow that's a big speed increase!!! 

image

**Comment by jywarren** [comment]
Let's keep an eye on these speeds. I hope this step has really helped!!!

**Comment by sagarpreet-chadha** [comment]
Wow 🎉

**Pull request merged**